\documentclass{article}
\usepackage{enumitem}
\usepackage{ amssymb }
\usepackage{ textcomp }
\usepackage{longtable}
\usepackage{amsmath,tabu}


\topmargin=-0.45in
\evensidemargin=0in
\oddsidemargin=0in
\textwidth=6.5in
\textheight=9.0in
\headsep=0.25in

\title{Perturb Seq}
\author{Caleb Lareau}
\date{\today}


\begin{document}
\maketitle

\begin{figure}[ht]
    \centering
    \includegraphics[width=\textwidth]{overview.png}
    \caption{\textbf{A graphical overview of the generalized linear model fit in Perturb-Seq analyses.} \newline \newline
    \label{fig:overview}}
\end{figure}

\section*{Introduction}
In three recent papers published in the journal \textit{Cell}, researchers at the Broad Institute, UCSF, and the Weizmann Institute described a new technology called Perturb-Seq, which integrates CRISPR-Cas9 genome editing and droplet-based single-cell RNA-Seq. \textbf{Figure \ref{fig:overview}} Additionally, a well-crafted Perturb-Seq experiment could enable the dissection gene regulation in the context of disease.


 Basically they use CRISPRi to knockdown different genes in different cells, then examine the transcriptome outcome of the different knockdowns through single-cell RNA-seq. Very often after a genome-wide CRISPR screen, you often have many hits, so they can use CRISPRi to validate each hits and explore the mechanism of the KD effect. Although single-cell readout from Drop-Seq could be noisy, they sequence ~500 cells per CRISPRi, so the average from the ~500 should give a much more robust expression readout. 

\begin{figure}[ht]
    \centering
    \includegraphics[width=\textwidth]{model.png}
    \caption{\textbf{A graphical overview of the generalized linear model fit in Perturb-Seq analyses.}}
     \label{fig:model}
\end{figure}


\section*{Where the stats are now}
Atray Dixit, the first author of the Perturb-Seq paper, maintains a \href{https://github.com/asncd/MIMOSCA}{Github repository}

\section*{Links to the BST245 Course}

\section*{Presentation Outline}
\begin{itemize}
  \item 0-6 minutes: discussion of CRISPR, scRNA-Seq, Drop-Seq, and requisite technological innovations as background
  \item 6-10 minutes: introduction of Perturb-Seq
  \item 
\end{itemize}


\section*{References}
Adamson, Britt, et al. "A multiplexed single-cell CRISPR screening platform enables systematic dissection of the unfolded protein response." Cell 167.7 (2016): 1867-1882. \newline \newline
Dixit, Atray, et al. "Perturb-seq: dissecting molecular circuits with scalable single-cell RNA profiling of pooled genetic screens." Cell 167.7 (2016): 1853-1866. \newline \newline
Jaitin, Diego Adhemar, et al. "Dissecting Immune Circuits by Linking CRISPR-Pooled Screens with Single-Cell RNA-Seq." Cell 167.7 (2016): 1883-1896. \newline \newline
Wagner, Daniel E., and Allon M. Klein. "Genetic screening enters the single-cell era." Nature Methods 14.3 (2017): 237-238.
\end{document}